### PR TITLE
Add RDS stream support

### PR DESCRIPTION
**Pull request by AlwinEsch**
With this request RDS support becomes added to mpeg2 audio on radio channels.

There on second patch becomes also a backword compatibility to use without RDS added.

### CODE REVIEW

File: parser_MPEGAudio.c [hunk 1]
```diff
@@ -82,9 +94,74 @@ void cParserMPEG2Audio::Parse(sStreamPacket *pkt)
...
+          }
+
+          m_RDSExtPID = m_Stream->AddExtDataType(scRDS);
+          if (!m_RDSExtPID)
+          {
+            ERRORLOG("PVR Parser MPEG2-Audio - %s - failed to add RDS data stream", __FUNCTION__);
+            m_RDSEnabled = false;
+            return;
+          }
+  }
```
Comment (FernetMenta) [inline parser_MPEGAudio.c]: indent

File: parser_MPEGAudio.c [hunk 2]
```diff
@@ -82,9 +94,74 @@ void cParserMPEG2Audio::Parse(sStreamPacket *pkt)
...
+
+        if (rdsl >= m_RDSBufferSize)
+        {
+          if (rdsl >= MAX_RDS_BUFFER_SIZE)
+          {
+            ERRORLOG("PVR Parser MPEG2-Audio - %s - max buffer size (%i kB) for RDS data reached, pid: %d", __FUNCTION__, MAX_RDS_BUFFER_SIZE/1024, m_pID);
+            m_RDSEnabled = false;
+            return;
+}
```
Comment (FernetMenta) [inline parser_MPEGAudio.c]: oops :)

File: streamer.c [hunk 3]
```diff
@@ -167,32 +168,53 @@ void cLiveStreamer::Close(void)
...
+          pkt_std_data.reftime = 0;
+        }
+        sendStreamPacket(&pkt_std_data);
+      }
+
+      // If some additional data is present inside the stream, it is written there (currently RDS inside MPEG2-Audio)
+      if (pkt_add_data.data)
       {
-        if (pkt.streamChange || requestStreamChange)
+        if (pkt_add_data.streamChange || requestStreamChangeExtData)
```
Comment (FernetMenta) [inline streamer.c]: the packet is names add_data, the other parameter ExtData. why not both add or both ext.
(ffmepg has named thos kind of things side_data)

**Comment by AlwinEsch** [comment]
Is updated, have now the sideData name like on ffmpeg.

**Comment by FernetMenta** [comment]
ok, merge at your will

File: streamer.c [hunk 4]
```diff
@@ -167,32 +168,53 @@ void cLiveStreamer::Close(void)
...
+      // If some additional data is present inside the stream, it is written there (currently RDS inside MPEG2-Audio)
+      if (pkt_side_data.data)
       {
-        if (pkt.streamChange || requestStreamChange)
+        if (pkt_side_data.streamChange || requestStreamChangeExtData)
           sendStreamChange();
-        requestStreamChange = false;
-        if (pkt.reftime)
+        requestStreamChangeExtData = false;
+        if (pkt_side_data.reftime)
```
Comment (FernetMenta) [inline streamer.c]: side_data does not need reftime, right?

Comment (AlwinEsch) [inline streamer.c]: Note in the moment but in the future comes maybe another additional data. I must look how is the best way to pass data about, for the future is HbbTV info data from DVB and if I find somewhere a radio channel which use aac to look what’s there on additional data.

Comment (FernetMenta) [inline streamer.c]: reftime is only sent once after start of streaming. sending it another time would ruin the entire function. reftime defines the time we hat at a particular timestamp.

Comment (AlwinEsch) [inline streamer.c]: For now it can be removed, let me check on stream after all new build are finished and I create a new pull request.

**Pull request merged**